### DISCUSSION TITLE
PoC of Dedicated Server finished

### DISCUSSION THREAD

**Discussion by Baldie-dev** [NebulaModTeam/nebula#54] [General]
Hi Guys,
I have started on the dedicated server and with very few modifications it is working ([commit](https://github.com/Baldie-dev/nebula/commit/112b36d214075c3d3b4c423077af1410f04682b8)). I will be still working on more improvements:
- Fix animation bug (animations are not being probably calculated on the server and it is kind of glitching on the clients)
- Do not load resources that the dedicated server does not need (audio, textures, etc.)
- Do not compute stuff that a dedicated server does not need (shaders, etc...)
- Fix all errors in the logs
- Possibility to run it from the .exe file, currently, it needs to be run from the Steam with parameters.

You can try it by running the server via steam with parameters "-server -batchmode -nographics" and then connect to it from the second steam instance.

Should I create a pull request once it is polished? or is it fine now? :) 

Thanks!

**Comment by hubastard** [comment]
Once it's more polished you could create a pull request. One thing that will need to be addressed is to be able to start the server from any savefile instead of creating a brand new game each time you start the server.

**Reply by Baldie-dev** [reply]
I think I have solved all major issues and I will create a pull request soon:

- No issues found, all synchronization seems to work properly.
- No exception is raised (due to the shaders computation or game logic) when the server starts in dedicated mode.
- Some drawing and texture loading methods are not executed to save CPU time.
- Server does not spawn local player if executed in dedicated mode.
- It is possible to start the server from the existing save file
- It is possible to run it from the command line:
`steam.exe -applaunch 1366540 -batchmode -nographics -server -load "asd"`

The only persisting issue is, when the game should start in windowed mode it will show only a white window and the server runs in the background. But it is working fine if game settings are set to run in fullscreen mode. No window is displayed and it fully runs in the background. I tried to force it to run it in fullscreen with param `-screen-fullscreen 1` but it does not help.
